streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -1390,6 +1390,46 @@ public void shouldNotCloseInternalProducerForNonEOS() {
         streamsProducer.flush();
     }
 
+    @Test
+    public void testCollectorFlush_ThrowsStreamsExceptionUsingDefaultExceptionHandler() {
+        final ErrorStringSerializer errorSerializer = new ErrorStringSerializer();
+        final RecordCollector collector = new RecordCollectorImpl(
+            logContext,
+            taskId,
+            streamsProducer,
+            new DefaultProductionExceptionHandler(),
+            streamsMetrics,
+            topology
+        );
+        collector.initialize();
+        collector.send(topic, "key", "val", null, 0, null, stringSerializer, errorSerializer, sinkNodeName, context);
+        assertThrows(StreamsException.class, collector::flush);
+    }
+
+    @Test
+    public void testCollectorFlush_DoesNotThrowStreamsExceptionUsingAlwaysContinueExceptionHandler() {
+        final ErrorStringSerializer errorSerializer = new ErrorStringSerializer();
+        final RecordCollector collector = new RecordCollectorImpl(
```

Comment:
nit: consider a method to return the `RecordCollector` and pass the exception hander as a parameter here and at line 1396.